Purchases/Purchasing/RCStoreKitRequestFetcher.h

```diff
@@ -15,20 +15,13 @@ typedef void(^RCFetchProductsCompletionHandler)(NSArray<SKProduct *> *products);
 
 typedef void(^RCFetchReceiptCompletionHandler)(void);
 
-@class SKProduct, SKProductsRequest;
-
-@interface RCProductsRequestFactory : NSObject
-- (SKProductsRequest *)requestForProductIdentifiers:(NSSet<NSString *> *)identifiers;
+@interface RCReceiptRefreshRequestFactory : NSObject
 - (SKReceiptRefreshRequest *)receiptRefreshRequest;
 @end
 
-@interface RCStoreKitRequestFetcher : NSObject <SKProductsRequestDelegate>
```

Comment:
not sure why the `SKProductsRequestDelegate` conformance was declared on the header, but it should be fine to have it in the `.m`

Reply:
is there some advantage to being more generic in the header?

Reply:
It's helpful if the interface is public so that the conformance is declared publicly incase somebody wanted to explicitly use RCStoreKitRequestFetcher as the delegate.

Reply:
Ah, that makes sense, I hadn't considered the idea of others using this as the delegate. 
I usually try to keep the headers as light as possible to restrict what anyone can do with the class to only the intended behavior. 
I don't have a strong preference here, though, so happy to make it public again if you think it's better

Reply:
Well, the class gets replaced with a swift implementation in any case in part 2, and in swift the protocol performance is forcibly public, so it wouldn't make much of a difference for this particular case. Happy to talk about handling these in a more general case, though

Reply:
Sounds good, we could definitely compose a new class, called the same thing (for backwards compatibility), but then encapsulate the swift version of this class as something like `StoreKitRequestFetcherPrivate` that we use. I don't understand well enough right now if we should do this.

Reply:
True. I don't think we'd get any value out of it, though, given that this is a fairly simple class, that just conforms to the delegate to receive the skRequests. FWIW doing the protocol conformance publicly would be the equivalent what what we had before the migration (well, once we have all the `public` stuff moved to `internal`, that is)

Reply:
Yeah, this also isn't exposed to the public, so I think you're good to go 👍